File: app/src/main/java/com/example/foodorderingapp/DetailsActivity.java
package com.example.foodorderingapp;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import com.example.foodorderingapp.databinding.ActivityDetailsBinding;

public class DetailsActivity extends AppCompatActivity {
    private ActivityDetailsBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityDetailsBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        // Retrieve data from the intent
        String foodName = getIntent().getStringExtra("MenuItemName");
        int foodImage = getIntent().getIntExtra("MenuItemImage", 0);

        // Set the food name and image
        binding.detailFoodName.setText(foodName);
        binding.detailFoodImage.setImageResource(foodImage);

        // Set click listener to finish the activity
        binding.imageButtonDetail.setOnClickListener(v -> finish());
    }
}

File: app/src/main/java/com/example/foodorderingapp/DetailsActivity.kt
package com.example.foodorderingapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.foodorderingapp.databinding.ActivityDetailsBinding

class DetailsActivity : AppCompatActivity() {
    private lateinit var binding: ActivityDetailsBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityDetailsBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val foodName = intent.getStringExtra("MenuItemName")
        val foodImage = intent.getIntExtra("MenuItemImage", 0)

        binding.detaillFoodName.text = foodName
        binding.detailFoodImage.setImageResource(foodImage)

        binding.imageButtonDetail.setOnClickListener {
            finish()
        }
    }
}